chill-scala/src/main/scala/com/twitter/chill/ScalaKryoInstantiator.scala

```diff
@@ -201,5 +202,13 @@ class AllScalaRegistrar extends IKryoRegistrar {
     val boxedUnit = scala.Unit.box(())
     k.register(boxedUnit.getClass, new SingletonSerializer(boxedUnit))
     PackageRegistrar.all()(k)
+
+    // Enable Java 8 lambda serialization only if we are running on a Java 8 JRE:
```

Comment:
I think we should make a `Java8ClosureRegistrar` that is an `IKryoRegistrar`: https://github.com/twitter/chill/blob/develop/chill-java/src/main/java/com/twitter/chill/IKryoRegistrar.java and it should be a no-op when we are not on java8 (as you have below).

See for instance the `PackageRegistrar.all()` above.